core/src/transport/memory.rs

```diff
@@ -192,7 +192,7 @@ impl Stream for Listener {
         };
         let event = ListenerEvent::Upgrade {
             upgrade: future::ok(channel),
-            listen_addr: self.addr.clone(),
+            local_addr: self.addr.clone().with(Protocol::Memory(self.port.get())),
```

Comment:
That might be a stupid question, but doesn't that produce `/memory/N/memory/M`?

Reply:
No, you are quite right and it was a confused attempt to come up with a local address. I reverted it back to `local_addr: self.addr.clone(),`. Thanks!